decision-records/2024-06-04--service-to-stop-while-waiting-for-cert.md

```diff
@@ -0,0 +1,106 @@
+**Date:** 2024-06-04<br/>
+**Author:** @sed-i
+
+## Context and Problem Statement
+When an admin forms a tls relation, there is a time window in which:
+
+1. the juju model tells us we must operate with TLS enabled; however
+2. we haven't received the server cert yet.
+
+As a result we had some code ordering issues (e.g. #414):
+- Should we update the config file with the tls config, which includes path to
+  the cert, before we got the cert, only to perfectly reflect the juju model?
+- Should we wait with the tls config and allow the workload to continue to
+  operate in plain http, in contradiction to the juju model?
+- How should we determine within the charm if "TLS is enabled"?
+
+This problem is common to many charms.
+
+## Decision
+A server shall NOT keep using plain http while waiting for a cert. The service
+must be stopped until the cert is in place.
+The charm shall have a "Waiting" status during that period (subject to other
```

Comment:
```suggestion
The charm shall surface this state to the admin by setting a "Waiting" status during that period (subject to other
```

Reply:
Either phrasing is probably correct, depending on [what you mean by "charm"](https://discourse.charmhub.io/t/what-do-we-mean-when-we-say-charm/8603).